src/AvatarPair.tsx

```diff
@@ -12,33 +13,37 @@ const Wrapper = styled('div')`
   position: relative;
 `
 
-const childStyles = props => ({
+const childStyles = (props: CSSObject) => ({
   display: 'inline-block',
   overflow: 'hidden', // Ensure page layout in Firefox should images fail to load
   lineHeight: `${get('lineHeights.condensedUltra')}`,
   verticalAlign: 'middle',
   borderRadius: '2px',
-  position: 'absolute',
+  position: 'absolute' as const,
   right: '-15%',
   bottom: '-9%',
   boxShadow: '-2px -2px 0 rgba(255,255,255,0.8)',
   backgroundColor: `${getBackgroundColor(props)}`
 })
 
 const ChildAvatar = styled(Avatar)(childStyles)
-const AvatarPair = ({children, ...rest}) => {
+const AvatarPair = ({children, ...rest}: PropsWithChildren<AvatarProps>) => {
   const avatars = React.Children.map(children, (child, i) => {
-    return i === 0 ? React.cloneElement(child, {size: 40}) : <ChildAvatar {...child.props} size={20} />
+    return i === 0 && React.isValidElement(child) ? (
+      React.cloneElement(child, {size: 40})
+    ) : (
+      <ChildAvatar {...(child as React.ReactElement<any>).props} size={20} />
+    )
   })
-  return <Wrapper {...rest}>{avatars}</Wrapper>
+  return <Wrapper {...(rest as any)}>{avatars}</Wrapper>
 }
 
 // styled() changes this
 AvatarPair.displayName = 'AvatarPair'
 
 AvatarPair.defaultProps = {theme}
 AvatarPair.propTypes = {
-  children: PropTypes.instanceOf(Avatar),
+  children: PropTypes.instanceOf(Avatar as any),
```

Comment:
We'll need to do this for any PropTypes that are ambiguous nodes or specific components correct?

Reply:
Because we're not that interested in type checking the PropTypes things, I didn't spend too long figuring out why these don't compile. I would expect `yes` is the answer to your question.

Reply:
Hello! 

I would love to contribute to this Pull Request.
Correct me if I'm wrong, but refactoring the components to typescript also remove the need for PropTypes, no?

Reply:
@thebiltheory We considered this question and decided PropTypes was good for consumers of Primer _not_ using TypeScript. If possible it would be nice to keep.

Reply:
@T-Hugs You are right, it makes sense to check props at runtime. However, it would be a double work to write an Interface + Proptypes object. 

Would you consider something like converting interfaces to proptypes?

For Example: 
https://github.com/milesj/babel-plugin-typescript-to-proptypes

Reply:
Hi, maybe it would make sense to use `type AvatarProps = PropTypes.InferProps<typeof AvatarPairPropTypes>;`? TypeScript can infer types from the `propTypes`. Instead of importing the global types, the prop-types could be used to generate the global/exported types.